src/irc/message.rs

```diff
@@ -1,11 +1,21 @@
+use crate::error::Error;
```

Comment:
is there a reason to have `error.rs` in `src` instead of in the `irc` directory,
I think it if was in the irc subdirectory then it would be clearer that it is part of the module
(but maybe I'm missing something)